src/index.ts

```diff
@@ -1471,7 +1471,7 @@ export class GossipSub extends EventEmitter<GossipsubEvents> implements PubSub<G
     const now = Date.now()
     let doPX = this.opts.doPX
 
-    graft.forEach(({ topicID }) => {
+    graft.forEach(async ({ topicID }) => {
```

Comment:
Having to allocate a promise for every single graft packet for this feature is not ideal

Reply:
Why is that?

Reply:
allocating promises has a non trivial cost in memory specially. We receive many grafts, so having to pay that cost for a feature that (at least) we don't use worsens performance for everyone. Ideally we want zero cost disabled features. Also returning promises to forEach is weird, as those are not handled.

Why don't you just do
```ts
for (const {topicID} of graft) {
```